front/src/modules/trainschedule/components/ManageTrainSchedule/SubmitConfAddTrainSchedule.tsx

```diff
@@ -32,65 +38,66 @@ export default function SubmitConfAddTrainSchedule({
   refetchTimetable,
   refetchConflicts,
   setIsWorking,
-}: Props) {
+}: SubmitConfAddTrainScheduleProps) {
   const [postTrainSchedule] =
     osrdEditoastApi.endpoints.postTrainScheduleStandaloneSimulation.useMutation();
   const dispatch = useDispatch();
   const { t } = useTranslation(['operationalStudies/manageTrainSchedule']);
 
+  const { getConf } = useOsrdSelectors() as OperationalStudiesConfSelectorsType;
+  const simulationConf = useSelector(getConf);
+  const {
+    pathfindingID,
+    timetableID,
+    departureTime,
+    trainCount,
+    trainDelta,
+    trainStep,
+    name: confName,
+  } = simulationConf;
+
   async function submitConfAddTrainSchedules() {
-    const { osrdconf } = store.getState();
-    const osrdConfig = formatConf(dispatch, t, osrdconf.simulationConf);
+    const osrdConfig = formatConf(dispatch, t, simulationConf);
 
-    if (!osrdconf.simulationConf.pathfindingID) {
+    if (!pathfindingID) {
       dispatch(
         setFailure({
           name: t('errorMessages.error'),
           message: t(`errorMessages.noPathfinding`),
         })
       );
-    } else if (
-      osrdConfig &&
-      osrdconf.simulationConf.pathfindingID &&
-      osrdconf.simulationConf.timetableID
-    ) {
+    } else if (osrdConfig && pathfindingID && timetableID) {
```

Comment:
We have already tested pathfindingId above
```suggestion
    } else if (osrdConfig && timetableID) {
```